### PR TITLE
Revert SQLite WAL journal change

### LINKED ISSUE #6193: sqlite exception unable to open database file
Updated the demo app to `4.2.0-beta.2` on Thursday and have had issues with crashing since. Clearing the app data will resolve the issue temporarily but after sometime, opening the app again and clicking an example will result in the crash again. I think this issue might have been introduced in PR https://github.com/mapbox/mapbox-gl-native/pull/5796.

```
08-29 09:57:17.098 29301-29789/? E/libc++abi: terminating with uncaught exception of type mapbox::sqlite::Exception: unable to open database file
08-29 09:57:17.098 29301-29856/? V/com.mapbox.mapboxsdk.http.HTTPRequest: [HTTP] Request was successful (code = 200).
                                                                          
                                                                          --------- beginning of crash
08-29 09:57:17.098 29301-29789/? A/libc: Fatal signal 6 (SIGABRT), code -6 in tid 29789 (DefaultFileSour)
                                         
                                         [ 08-29 09:57:17.102   365:  365 W/         ]
                                         debuggerd: handling request: pid=29301 uid=10307 gid=10307 tid=29789
08-29 09:57:17.107 29301-29301/? D/mbgl: {pboxandroiddemo}[Android]: NativeMapView::invalidate()
08-29 09:57:17.187 29857-29857/? A/DEBUG: *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
08-29 09:57:17.187 29857-29857/? A/DEBUG: Build fingerprint: 'google/bullhead/bullhead:7.0/NRD90M/3085278:user/release-keys'
08-29 09:57:17.187 29857-29857/? A/DEBUG: Revision: 'rev_1.0'
08-29 09:57:17.187 29857-29857/? A/DEBUG: ABI: 'arm64'
08-29 09:57:17.187 29857-29857/? A/DEBUG: pid: 29301, tid: 29789, name: DefaultFileSour  >>> com.mapbox.mapboxandroiddemo <<<
08-29 09:57:17.187 29857-29857/? A/DEBUG: signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
08-29 09:57:17.193 29857-29857/? A/DEBUG: Abort message: 'terminating with uncaught exception of type mapbox::sqlite::Exception: unable to open database file'
08-29 09:57:17.193 29857-29857/? A/DEBUG:     x0   0000000000000000  x1   000000000000745d  x2   0000000000000006  x3   0000000000000008
08-29 09:57:17.194 29857-29857/? A/DEBUG:     x4   ffffffffffffffff  x5   0000000000000000  x6   0000008000808080  x7   651f647260616073
08-29 09:57:17.194 29857-29857/? A/DEBUG:     x8   0000000000000083  x9   ffffffffffffffdf  x10  0000000000000000  x11  0000000000000001
08-29 09:57:17.194 29857-29857/? A/DEBUG:     x12  0000000000000018  x13  0000000000000000  x14  0000000000000000  x15  0005e11a50794557
08-29 09:57:17.194 29857-29857/? A/DEBUG:     x16  0000007d739e1ed0  x17  0000007d7398ba2c  x18  0000000000000000  x19  0000007d53e634f8
08-29 09:57:17.194 29857-29857/? A/DEBUG:     x20  0000000000000006  x21  0000007d53e63450  x22  000000000000001e  x23  00000000ffffffc8
08-29 09:57:17.194 29857-29857/? A/DEBUG:     x24  0000007d53e62630  x25  0000007d53e62500  x26  0000007d53e62540  x27  00000000000f4240
08-29 09:57:17.194 29857-29857/? A/DEBUG:     x28  0000007d53e62da8  x29  0000007d53e62410  x30  0000007d73988e58
08-29 09:57:17.194 29857-29857/? A/DEBUG:     sp   0000007d53e623f0  pc   0000007d7398ba34  pstate 0000000060000000
08-29 09:57:17.207 29857-29857/? A/DEBUG: backtrace:
08-29 09:57:17.208 29857-29857/? A/DEBUG:     #00 pc 000000000006ba34  /system/lib64/libc.so (tgkill+8)
08-29 09:57:17.208 29857-29857/? A/DEBUG:     #01 pc 0000000000068e54  /system/lib64/libc.so (pthread_kill+64)
08-29 09:57:17.208 29857-29857/? A/DEBUG:     #02 pc 0000000000023ed8  /system/lib64/libc.so (raise+24)
08-29 09:57:17.208 29857-29857/? A/DEBUG:     #03 pc 000000000001c790  /system/lib64/libc.so (abort+52)
08-29 09:57:17.208 29857-29857/? A/DEBUG:     #04 pc 000000000035d238  /data/app/com.mapbox.mapboxandroiddemo-1/lib/arm64/libmapbox-gl.so
08-29 09:57:17.208 29857-29857/? A/DEBUG:     #05 pc 000000000035d348  /data/app/com.mapbox.mapboxandroiddemo-1/lib/arm64/libmapbox-gl.so
08-29 09:57:17.208 29857-29857/? A/DEBUG:     #06 pc 000000000035ab38  /data/app/com.mapbox.mapboxandroiddemo-1/lib/arm64/libmapbox-gl.so
08-29 09:57:17.208 29857-29857/? A/DEBUG:     #07 pc 000000000035a1dc  /data/app/com.mapbox.mapboxandroiddemo-1/lib/arm64/libmapbox-gl.so (__cxa_throw+204)
08-29 09:57:17.208 29857-29857/? A/DEBUG:     #08 pc 0000000000161468  /data/app/com.mapbox.mapboxandroiddemo-1/lib/arm64/libmapbox-gl.so
08-29 09:57:17.208 29857-29857/? A/DEBUG:     #09 pc 000000000015b8d0  /data/app/com.mapbox.mapboxandroiddemo-1/lib/arm64/libmapbox-gl.so
08-29 09:57:17.208 29857-29857/? A/DEBUG:     #10 pc 000000000015af3c  /data/app/com.mapbox.mapboxandroiddemo-1/lib/arm64/libmapbox-gl.so
08-29 09:57:17.208 29857-29857/? A/DEBUG:     #11 pc 0000000000151854  /data/app/com.mapbox.mapboxandroiddemo-1/lib/arm64/libmapbox-gl.so
08-29 09:57:17.208 29857-29857/? A/DEBUG:     #12 pc 0000000000151730  /data/app/com.mapbox.mapboxandroiddemo-1/lib/arm64/libmapbox-gl.so
08-29 09:57:17.208 29857-29857/? A/DEBUG:     #13 pc 0000000000154ff8  /data/app/com.mapbox.mapboxandroiddemo-1/lib/arm64/libmapbox-gl.so
08-29 09:57:17.208 29857-29857/? A/DEBUG:     #14 pc 0000000000155df8  /data/app/com.mapbox.mapboxandroiddemo-1/lib/arm64/libmapbox-gl.so
08-29 09:57:17.208 29857-29857/? A/DEBUG:     #15 pc 0000000000155c98  /data/app/com.mapbox.mapboxandroiddemo-1/lib/arm64/libmapbox-gl.so
08-29 09:57:17.209 29857-29857/? A/DEBUG:     #16 pc 000000000014a684  /data/app/com.mapbox.mapboxandroiddemo-1/lib/arm64/libmapbox-gl.so
08-29 09:57:17.209 29857-29857/? A/DEBUG:     #17 pc 0000000000146a54  /data/app/com.mapbox.mapboxandroiddemo-1/lib/arm64/libmapbox-gl.so
08-29 09:57:17.209 29857-29857/? A/DEBUG:     #18 pc 0000000000146f6c  /data/app/com.mapbox.mapboxandroiddemo-1/lib/arm64/libmapbox-gl.so
08-29 09:57:17.209 29857-29857/? A/DEBUG:     #19 pc 000000000014f490  /data/app/com.mapbox.mapboxandroiddemo-1/lib/arm64/libmapbox-gl.so
08-29 09:57:17.209 29857-29857/? A/DEBUG:     #20 pc 000000000014f3e0  /data/app/com.mapbox.mapboxandroiddemo-1/lib/arm64/libmapbox-gl.so
08-29 09:57:17.209 29857-29857/? A/DEBUG:     #21 pc 000000000006863c  /system/lib64/libc.so (_ZL15__pthread_startPv+208)
08-29 09:57:17.209 29857-29857/? A/DEBUG:     #22 pc 000000000001d9fc  /system/lib64/libc.so (__start_thread+16)
```

cc: @mapbox/android

### PR DESCRIPTION
**Pull request by friedbunny**
Reverts https://github.com/mapbox/mapbox-gl-native/pull/5796, which changed the SQLite database’s journal mode to Write-Ahead Logging (WAL). This was a performance increase, but brought instability to Android (#6193).

`journal_mode = DELETE` and `synchronous = FULL` are the system defaults, but they’re now also set explicitly so that there’s no potential confusion between the default schema and the v5 migration.

Fixes #6193 and obviates #6319.

/cc @jfirebaugh @zugaldia

### CODE REVIEW

File: platform/default/mbgl/storage/offline_database.cpp [hunk 1]
```diff
@@ -49,7 +49,8 @@ void OfflineDatabase::ensureSchema() {
             case 1: break; // cache-only database; ok to delete
             case 2: migrateToVersion3(); // fall through
             case 3: migrateToVersion4(); // fall through
```
Comment (jfirebaugh) [inline platform/default/mbgl/storage/offline_database.cpp]: Thinking out loud: we could remove `OfflineDatabase::migrateToVersion4`, and have `case 3` fall through to `migrateToVersion5()` without doing anything itself. But then we'd have two types of databases in the wild: those that have been switched temporarily to WAL and back, and those that haven't. There _probably_ isn't any interesting differences between those two types, but why introduce the variable?

OTOH, what if switching to WAL, even temporarily, can causes an issue in some corner case? Then we _would_ be better off avoiding the migration through v4 when starting with v3.

I don't see a strong argument either way, so I'm fine with keeping this as is here.

Comment (1ec5) [inline platform/default/mbgl/storage/offline_database.cpp]: Would it be feasible to smoketest a collection of test fixture databases that go through different subsets of these migration paths?

Comment (jfirebaugh) [inline platform/default/mbgl/storage/offline_database.cpp]: Yes. In fact, we should require that every new migration be accompanied by the addition of a test case that tests migrating from the prior version. In local development, migrations are a rarely-exercised code path.

Comment (friedbunny) [inline platform/default/mbgl/storage/offline_database.cpp]: That’s a great point — I’d lean towards removing `migrateToVersion4` and eliminating any possibility of journal weirdness for the vast majority of users.

Comment (friedbunny) [inline platform/default/mbgl/storage/offline_database.cpp]: I’ll look at adding migration tests.

Comment (jfirebaugh) [inline platform/default/mbgl/storage/offline_database.cpp]: I'm fine with that too.

Can you create `MigrateFromV3Schema` and `MigrateFromV4Schema` testcases, using checked-in database fixtures generated by upgrading the existing `test/fixtures/offline_database/v2.db` fixture to v3 and then v4 prior to this PR?

**Comment by friedbunny** [comment]
- Added basic migration tests for v3→v5 and v4→v5, including database fixtures.
- Removed the v4 migration path, per discussion.

**Comment by friedbunny** [comment]
Android CI failures are unrelated to these changes, but I’ll wait for those fixes to land before rebasing/pushing for another chance at the 💚. Otherwise, this is ready for review.

File: test/storage/offline_database.cpp [hunk 2]
```diff
@@ -554,23 +554,85 @@ static int databaseUserVersion(const std::string& path) {
...
+    {
+        OfflineDatabase db("test/fixtures/offline_database/v5.db", 0);
+        auto regions = db.listRegions();
+        for (auto& region : regions) {
+            db.deleteRegion(std::move(region));
+        }
+    }
+
+    EXPECT_EQ(5, databaseUserVersion("test/fixtures/offline_database/v5.db"));
+    EXPECT_LT(databasePageCount("test/fixtures/offline_database/v5.db"),
```
Comment (jfirebaugh) [inline test/storage/offline_database.cpp]: ✂️ this assertion (and in `MigrateFromV4Schema` too). It's specific to the v2 ⇢ v3 migration (checking that it runs `VACUUM`). If it passes for other migrations, it's coincidental.

Comment (friedbunny) [inline test/storage/offline_database.cpp]: Done. 👍

**Pull request merged**